presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -542,15 +632,61 @@ public Duration getTotalCpuTime()
     public void start()
     {
         if (started.compareAndSet(false, true)) {
-            executor.submit(this::schedule);
+            startScheduling(getStagesReadyForExecution());
+        }
+    }
+
+    private List<SqlStageExecution> getStagesReadyForExecution()
+    {
+        return stream(forTree(StreamingPlanSection::getChildren).depthFirstPreOrder(sectionedPlan))
+                .filter(this::isReadyForExecution)
+                .flatMap(section -> stream(forTree(SubPlan::getChildren).depthFirstPreOrder(section.getPlan())))
```

Comment:
Are we still schedule all the sections that are ready?

Reply:
That is something the `PhasedExecutionStrategy` has to control. However apparently it doesn't. I'm going to dig deeper into it, and if needed - open a separate PR.